lib/retrieve_title.rb

```diff
@@ -27,6 +27,7 @@ def self.extract_title(html, encoding = nil)
     end
 
     if title.present?
+      return nil if title =~ /<\/t/
```

Comment:
this feels a bit odd, can you expand on this change? why is it needed? can we add a test?

Reply:
This is needed because of this condition:

https://github.com/discourse/discourse/blob/1472e47aae5bfdfb6fd9abfe89beb186c751f514/lib/retrieve_title.rb#L78

Since Nokogiri fixes the broken HTML tags the additional HTML chunks are not received and it results in broken title tag (the one with `</titl` in title text). We need to confirm that the extracted title is not from the broken tag hence I'm checking for presence of `</t` in title text.

Adding a text here is tricky because to test this case we need to stream chunks of HTML from a URL and need to break the chunk just when the title tag appears.

Reply:
I added a test for this change.